runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java

```diff
@@ -0,0 +1,1266 @@
+            Flyweight extension)
+        {
+            replySeq = sequence;
+            replyAck = acknowledge;
+
+            doData(sender, originId, routedId, replyId, replySeq, replyAck, replyMax,
+                traceId, authorization, flags, replyBud, reserved, payload, extension);
+        }
+
+        private void doStreamEnd(
+            long traceId)
+        {
+            if (!KafkaState.replyClosed(state))
+            {
+                state = KafkaState.closedReply(state);
+
+                doEnd(sender, originId, routedId, replyId, 0, 0, 0,
+                    traceId, authorization, EMPTY_EXTENSION);
+
+                streamsByInitialIds.remove(initialId);
+            }
+        }
+
+        private void doStreamAbort(
+            long traceId)
+        {
+            if (!KafkaState.replyClosed(state))
+            {
+                state = KafkaState.closedReply(state);
+
+                doAbort(sender, originId, routedId, replyId, 0, 0, 0,
+                    traceId, authorization, EMPTY_EXTENSION);
+
+                streamsByInitialIds.remove(initialId);
+            }
+        }
+
+        private void doStreamReset(
+            long traceId)
+        {
+            if (!KafkaState.initialClosed(state))
+            {
+                state = KafkaState.closedInitial(state);
+
+                doReset(sender, originId, routedId, initialId, 0, 0, 0,
+                    traceId, authorization);
+
+                streamsByInitialIds.remove(initialId);
+            }
+        }
+
+        private void cleanupStream(
+            long traceId)
+        {
+            doStreamReset(traceId);
+            doStreamAbort(traceId);
+
+            streamsByInitialIds.remove(initialId);
+        }
+
+        private void signalNow(
+            int signalId)
+        {
+            connection.doConnectionSignalNow(initialId, signalId);
+        }
+
+        private long signalAt(
+            long timeMillis,
+            int signalId)
+        {
+            return connection.doConnectionSignalAt(initialId, timeMillis, signalId);
+        }
+
+        private void onSignal(
+            SignalFW signal)
+        {
+            final long traceId = signal.traceId();
+            final int signalId = signal.signalId();
+
+            if (signalId == SIGNAL_STREAM_REPLY_BEGIN)
+            {
+                doStreamBegin(traceId);
+            }
+            else if (signalId == SIGNAL_STREAM_REPLY_END)
+            {
+                doStreamEnd(traceId);
+            }
+            else if (signalId == SIGNAL_STREAM_REPLY_ABORT)
+            {
+                doStreamAbort(traceId);
+            }
+            else if (signalId == SIGNAL_STREAM_INITIAL_RESET)
+            {
+                doStreamReset(traceId);
+            }
+            else
+            {
+                doSignal(sender, originId, routedId, initialId, initialSeq, initialAck, initialMax, traceId, signalId);
+            }
```

Comment:
Change to switch statement on signalId with case per constant plus default for else clause.